asdf/tags/core/tests/test_ndarray.py

```diff
@@ -874,16 +874,15 @@ def test_readonly(tmpdir):
     # Opening in read mode (the default) should mean array is readonly
     with asdf.open(tmpfile) as af:
         assert af["data"].flags.writeable is False
-        with pytest.raises(ValueError) as err:
+        with pytest.raises(ValueError, match=r"assignment destination is read-only") as err:
             af["data"][0] = 41
             assert str(err) == "assignment destination is read-only"
```

Comment:
This line is actually unreachable as the error will occur on 878 and the context will end there. If you remove this line you should also remove the `as err` on 877 too

Reply:
Thanks for finding the issues. I updated the test code removing the `as err` lines and the unreachable line.